website/static/js/components/quickFiles.js

```diff
@@ -0,0 +1,172 @@
+                                }
+                            }, ctrl.file.attributes.name),
+                        ]),
+                        m('div', {style: 'width: 50%; float:right; font-size:small; line-height:2;'}, 'downloads: ' + ctrl.file.attributes.extra.downloads)
+                    ])
+                ])
+            ])
+        ]);
+    }
+};
+
+var QuickFiles = {
+
+    controller: function (options) {
+        var self = this;
+        self.user = options.user._id;
+        self.isProfile = options.user.is_profile;
+
+        self.quickFiles = m.prop([]);
+        self.requestPending = m.prop(false);
+
+        self.failed = false;
+        self.paginators = m.prop([]);
+        self.nextPage = m.prop('');
+        self.prevPage = m.prop('');
+        self.totalPages = m.prop(0);
+        self.currentPage = m.prop(0);
+        self.pageToGet = m.prop(0);
+
+        self.getQuickFiles = function _getQuickFiles(url) {
+            if (self.requestPending()) {
+                return;
+            }
+            self.quickFiles([]);
+            self.requestPending(true);
+
+            function _processResults(result) {
+
+                self.quickFiles(result.data);
+                self.nextPage(result.links.next);
+                self.prevPage(result.links.prev);
+
+                var params = $osf.urlParams(url);
+                var page = params.page || 1;
+
+                self.currentPage(parseInt(page));
+                self.totalPages(Math.ceil(result.meta.total / result.meta.per_page));
+
+                m.redraw();
+            }
+
+            var promise = m.request({
+                method: 'GET',
+                url: url,
+                background: true,
+                config: mHelpers.apiV2Config({withCredentials: window.contextVars.isOnRootDomain})
+            });
+
+            promise.then(
+                function (result) {
+                    self.requestPending(false);
+                    _processResults(result);
+                    return promise;
+                }, function (xhr, textStatus, error) {
+                    self.failed = true;
+                    self.requestPending(false);
+                    m.redraw();
+                    Raven.captureMessage('Error retrieving quickfiles', {
+                        extra: {
+                            url: url,
+                            textStatus: textStatus,
+                            error: error
+                        }
+                    });
+                }
+            );
+        };
+
+        self.getCurrentQuickFiles = function _getCurrentQuickFiles(page) {
+            if (!self.requestPending()) {
+                var url = _buildUrl(page, self.user);
+                return self.getQuickFiles(url);
+            }
+        };
+        self.getCurrentQuickFiles();
+    },
+
+    view: function (ctrl) {
+
+        profilePagination.ProfilePagination(ctrl, _buildUrl);
```

Comment:
+1 to making pagination re-usable. 

However, I suggest tweaking the approach a bit.

Notice that `ProfilePagination` function can't really be used on its own--it assumes a lot about what's on `ctrl` (i.e. it assumes that `ctrl` has props names `paginators`, `nextPage`, etc.).

Can you encapsulate *all* pagination-related code into one function?

I imagine an API like:

```js
var withPagination = require('js/components/pagination').withPagination;

function buildURL(...) {...}

// withPagination is a higher-order component, i.e. it takes a component as input
// and outputs a new component with added behavior
var QuickFiles = withPagination(buildURL, {
    controller: function() {
         // only quick files stuff here, no pagination stuff
     },
    view: function(ctrl) {
        // only quick files stuff here, no pagination stuff
     },
});
```